pkg/apis/policy/v1alpha1/propagation_types.go

```diff
@@ -173,6 +173,12 @@ type PropagationSpec struct {
 	// +kubebuilder:validation:Enum=Lazy
 	// +optional
 	ActivationPreference ActivationPreference `json:"activationPreference,omitempty"`
+
+	// OrphaningDeletion  represents whether to prevent removal of these managed resources
+	// +kubebuilder:default="Never"
+	// +kubebuilder:validation:Enum=Never;Complete
+	// +optional
+	OrphaningDeletionPolicy OrphaningDeletionPolicyType `json:"orphaningDeletionPolicy,omitempty"`
```

Comment:
I believe we only need to add this field in `PP`. The `work` does not require this field.

In addition, I don't think the current implementation can fully handle all situations.

Here's a new idea:

When users set `orphaningDeletionPolicy` of PP as `Complete`.

The specific implementation should be adding an `annotation` on the `workloadTemplat`e of the control plane, similar to `orphaningDeletionPolicy.karmada.io: "Complete"`

Then when deleting `workloadTemplate` in `execution_controller`, check if `workloadTemplate` has annotation `orphaningDeletionPolicy.karmada.io`. If it is `Complete`, do not delete it.

The benefits are:
1. The personnel creating the `workloadTemplate` can also control whether this application needs to be retained in member clusters, rather than only those who create PP.
2. Resources like namespace that do not have `PP` can also be deleted on the control plane without being deleted from member clusters.

Reply:
> Then when deleting workloadTemplate in execution_controller, check if workloadTemplate has annotation orphaningDeletionPolicy.karmada.io. If it is Complete, do not delete it.

Execution controller must watch all origin resource such as  resourceTemlate 、namespace and so on, which make too complicated, I don't think that's a good idea.
I think for namespace, can add annotation like `orphaningDeletionPolicy.karmada.io: "Complete"`; FederatedResourceQuota those create work object direct,  by adding field `orphaningDeletionPolicy: Complete` if necessary

Reply:
> Execution controller must watch all origin resource such as resourceTemlate 、namespace and so on

There's no need, this is existing logic.

Reply:
This is no different from directly adding annotations to the resource template. In addition, users can directly skip pp to control the annotation value on the resource template.

Reply:
> This is no different from directly adding annotations to the resource template. In addition, users can directly skip pp to control the annotation value on the resource template.



Yes, I think that's reasonable.

Reply:
> I believe we only need to add this field in PP. The work does not require this field.

+1, but I prefer to decouple from PP/CPP, cause I'm afraid this is very unfriendly to administrators if the system also has the `PropagationPolicyPreemption` feature turned on. in addition, I feel that propagation policies are becoming more and more complex, and almost all new features must consider adding APIs to it.

Reply:
> but I prefer to decouple from PP/CPP, cause I'm afraid this is very unfriendly to administrators if the system also has the PropagationPolicyPreemption feature turned on.

Can you give an example of this?

Reply:
![orphaning-delete](https://github.com/karmada-io/karmada/assets/89241565/46ad1854-36c2-4e0f-a3cb-2d14ee26e58d)

if decoupled from PP/CPP, the administrator don't care about it.

Reply:
Thanks, I get your meaning.  How do you think @CharlesQQ @chaunceyjiang @RainbowMango

Reply:
> but I prefer to decouple from PP/CPP, cause I'm afraid this is very unfriendly to administrators if the system also has the PropagationPolicyPreemption feature turned on.

I do not quite understand how `unfriendly` here, I think the preemption is clear that after a PropagationPolicy(with higher priority) successfully preempted a resource, then the resource should be propagated as per the PropagationPolicy.